druid/custom/rich_input/rich_input.lua

```diff
@@ -71,15 +140,26 @@ function RichInput.init(self, template, nodes)
 	self.druid = self:get_druid()
 	self.input = self.druid:new_input(self:get_node(SCHEME.BUTTON), self:get_node(SCHEME.INPUT))
 	self.cursor = self:get_node(SCHEME.CURSOR)
-
-	self.input:set_text("")
+	self.highlight = self:get_node(SCHEME.HIGHLIGHT) 
+	
 	self.placeholder = self.druid:new_text(self:get_node(SCHEME.PLACEHOLDER))
+	self.text = self.druid:new_text(self:get_node(SCHEME.INPUT))
 
 	self.input.on_input_text:subscribe(update_text)
 	self.input.on_input_select:subscribe(on_select)
 	self.input.on_input_unselect:subscribe(on_unselect)
-	on_unselect(self)
-	update_text(self, "")
+	
+	self.input.button.on_click:subscribe(on_button_click, self)
+	self.input.button.on_double_click:subscribe(on_button_double_click, self)
+	self.input.style.NO_CONSUME_INPUT_WHILE_SELECTED = true
```

Comment:
I think the rich input should not change these style settings (and for skip input keys same)